src/main/kotlin/de/cyface/uploader/DefaultUploader.kt

```diff
@@ -86,21 +84,22 @@ class DefaultUploader(private val apiEndpoint: String) : Uploader {
     override fun uploadAttachment(
         jwtToken: String,
         metaData: RequestMetaData,
-        measurementId: Long,
         file: File,
         fileName: String,
-        progressListener: UploadProgressListener
+        progressListener: UploadProgressListener,
     ): Result {
-        val endpoint = attachmentsEndpoint(measurementId)
+        val measurementId = metaData.measurementIdentifier.toLong()
+        val deviceId = metaData.deviceIdentifier
+        val endpoint = attachmentsEndpoint(deviceId, measurementId)
         return uploadFile(jwtToken, metaData, file, endpoint, progressListener)
     }
 
     override fun measurementsEndpoint(): URL {
         return URL(returnUrlWithTrailingSlash(apiEndpoint) + "measurements")
     }
 
-    override fun attachmentsEndpoint(measurementId: Long): URL {
-        return URL(returnUrlWithTrailingSlash(apiEndpoint) + "measurements/$measurementId/attachments")
+    override fun attachmentsEndpoint(deviceId: String, measurementId: Long): URL {
+        return URL(returnUrlWithTrailingSlash(apiEndpoint) + "measurements/$deviceId/$measurementId/attachments")
```

Comment:
this ensures the path is identical to the measurement path in the `provider` API, where we can delete measurements. this does not affect other clients as the RFR UI is the first where we use the DefaultUploader.attachments functionality. Digural has an own uploader Implementation and different path structures: `/devices/did/measurements/mid/attachments`. This makes sense there but here it we want to keep compatible with the current APIs.